pod/video/templatetags/video_tags.py

```diff
@@ -55,15 +55,22 @@ def get_last_videos(context):
     request = context['request']
     videos = Video.objects.filter(
         encoding_in_progress=False, is_draft=False,
-        sites=get_current_site(request))
+        sites=get_current_site(request)).exclude(
+            channel__visible=0)
 
     if not HOMEPAGE_SHOWS_PASSWORDED:
         videos = videos.filter(
             Q(password='') | Q(password__isnull=True))
     if not HOMEPAGE_SHOWS_RESTRICTED:
         videos = videos.filter(is_restricted=False)
 
-    videos = videos.exclude(
-        pk__in=[vid.id for vid in videos if not vid.encoded])
+    count = 0
+    recent_vids = []
+    for vid in videos:
+        if(vid.encoded):
+            recent_vids.append(vid)
+            count = count+1
+        if(count >= 12):
+            break
```

Comment:
il faudrait préciser un orderby non ? (il me semble qu'il y en a un dans le model mais la, tu récupères les 12 plus anciennes ou les 12 plus récentes ?)

Reply:
Justement justement je me posais la question, dans la version d'avant il n'y en a pas

Reply:
Alors je viens de tester et c'est assez incroyable : 
`from django.conf import settings as django_settings
from django import template
from django.utils.text import capfirst
from django.urls import reverse
from django.db.models import Q
from django.contrib.sites.shortcuts import get_current_site

from pod.video.models import Video

from datetime import datetime

HOMEPAGE_SHOWS_PASSWORDED = getattr(
    django_settings,
    'HOMEPAGE_SHOWS_PASSWORDED',
    True)
HOMEPAGE_SHOWS_RESTRICTED = getattr(
    django_settings,
    'HOMEPAGE_SHOWS_RESTRICTED',
    True)

def get_last_videos_1():
    request = None
    videos = Video.objects.filter(
        encoding_in_progress=False, is_draft=False,
        sites=get_current_site(request)).exclude(
            channel__visible=0)
    if not HOMEPAGE_SHOWS_PASSWORDED:
        videos = videos.filter(
            Q(password='') | Q(password__isnull=True))
    if not HOMEPAGE_SHOWS_RESTRICTED:
        videos = videos.filter(is_restricted=False)
    count = 0
    recent_vids = []
    for vid in videos:
        if(vid.encoded):
            recent_vids.append(vid)
            count = count+1
        if(count >= 12):
            break
    return recent_vids


def get_last_videos_2():
    request = None
    videos = Video.objects.filter(
        encoding_in_progress=False, is_draft=False,
        sites=get_current_site(request))
    if not HOMEPAGE_SHOWS_PASSWORDED:
        videos = videos.filter(
            Q(password='') | Q(password__isnull=True))
    if not HOMEPAGE_SHOWS_RESTRICTED:
        videos = videos.filter(is_restricted=False)
    videos = videos.exclude(
        pk__in=[vid.id for vid in videos if not vid.encoded])
    return videos.exclude(channel__visible=0)[:12]

def check_vid():
    print(datetime.now())
    videos = get_last_videos_1()
    for vid in videos:
        print(vid.title)
    print(datetime.now())
    videos = get_last_videos_2()
    for vid in videos:
        print(vid.title)
    print(datetime.now())
`
Et voici ce que ca donne : 
`check_vid()
2020-04-30 11:58:08.118180
Test 4 questions à Benedicte Gauthier
Interview Sylviane Pied
UE 10 Vidéo
Mme Sylviane Pied
UE 10
Introduction à Matlab_partie5
Introduction à Matlab_partie4
Introduction à Matlab_partie3
Introduction à Matlab_partie2
Intoduction à Matlab_partie1
Interview du PR. Philippe MARCHETTI, UE 10 : initiation à la recherche.
VR THE NEW REALITY?
2020-04-30 11:58:08.519006
Test 4 questions à Benedicte Gauthier
Interview Sylviane Pied
UE 10 Vidéo
Mme Sylviane Pied
UE 10
Introduction à Matlab_partie5
Introduction à Matlab_partie4
Introduction à Matlab_partie3
Introduction à Matlab_partie2
Intoduction à Matlab_partie1
Interview du PR. Philippe MARCHETTI, UE 10 : initiation à la recherche.
VR THE NEW REALITY?
2020-04-30 11:58:13.786611`

Reply:
A peine une demi seconde pour ta methode contre 5 secondes pour la mienne !

Reply:
ok, je viens de tester et je crois qu'en ajoutant ceci videos = videos.defer("video", "slug", "owner", "additional_owners", "description") avant le count = 0 on gagne encore plus de temps (peut-etre même qu'il faudrait le rajouter dans views.py de video et voir si tout fonctionne